x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/UsernamesField.java

```diff
@@ -16,6 +16,8 @@ public final class UsernamesField {
     public static final String SYSTEM_ROLE = "_system";
     public static final String XPACK_SECURITY_NAME = "_xpack_security";
     public static final String XPACK_SECURITY_ROLE = "_xpack_security";
+    public static final String SECURITY_PROFILE_NAME = "_security_profile";
```

Comment:
This is a great point! I thought about that one can keep using any existing `_security_profile` user without it clashing with the new internal user because role resolving for internal users has its own distinct path. But you are right that adding a new internal user has more impact on the user-facing side:
1. It will no longer be possible to add a new native realm user named `_security_profile`
2. If there is an existing `_security_profile` native realm user, it will not be possible to update it, including changing password or deleting it. But otherwise it will work normally.

@tvernum and I discussed and we don't think it should be considered breaking because we never considered adding new internal users breaking in the past. We don't want to change the existing stance because it means we will never be able to add another internal users.

That said, there is a discussion about how much validation we should have for username and more broadly the future of internal users. Today, you can create a `_xpack_security` using the file realm. This usename can also comes with external realms (LDAP etc) where we have no control over usernames. So why should native realm specifically bother with it especially since internally an "externally" defined `_xpack_security` will not be confused with the true internal user? 

In summary, the plan I will go with is:
1. Keep this PR as is.
2. Start a separate discussion about options to make adding internal users less impactful (aiming for 8.3 as a follow-up for this PR) and future direction of internal users.

Reply:
Agreed with the next steps! Adding my thoughts to the slack thread https://elastic.slack.com/archives/CA6N3HDF1/p1650527946891839